File: 2015/kt/src/test/kotlin/org/adventofcode/Day08Test.kt
package org.adventofcode

import org.junit.Assert.*
import org.junit.Test

class Day08Test {
  private val d: Day08 = Day08()
  private val lines = parseFile("08")

  @Test
  fun day08Diff() {
    assertEquals(2, d.diff(""""""""))
    assertEquals(2, d.diff(""""abc""""))
    assertEquals(3, d.diff(""""aaa\"aaa""""))
    assertEquals(5, d.diff(""""\x27""""))
    assertEquals(5, d.diff(""""\xfb""""))
    assertEquals(4, d.diff(""""d\\fb\"u""""))
  }

  @Test
  fun day08part1() {
    assertEquals(1342, d.part1(lines))
  }

  @Test
  fun day08Diff2() {
    assertEquals(4, d.diff2(""""""""))
    assertEquals(4, d.diff2(""""abc""""))
    assertEquals(6, d.diff2(""""aaa\"aaa""""))
    assertEquals(5, d.diff(""""\x27""""))
  }

  @Test
  fun day08part2() {
    assertEquals(2074, d.part2(lines))
  }
}

File: 2015/js/day08.test.js
import { diff, part1, diff2, part2 } from './day08'
import read from './read'

describe('2015 Day 08', () => {

  const lines = read('08')

  it('part1 example', () => {
    expect(diff(`""`)).toEqual(2)
    expect(diff(`"abc"`)).toEqual(2)
    expect(diff(`"aaa\\"aaa"`)).toEqual(3)
    expect(diff(`"\\x27"`)).toEqual(5)
  })

  it('part1', () => {
    expect(part1(lines)).toEqual(1342)
  })

  it('part2 example', () => {
    expect(diff2(`""`)).toEqual(4)
    expect(diff2(`"abc"`)).toEqual(4)
    expect(diff2(`"aaa\\"aaa"`)).toEqual(6)
    expect(diff(`"\\x27"`)).toEqual(5)
  })

  it('part2', () => {
    expect(part2(lines)).toEqual(2074)
  })

})
